### DISCUSSION TITLE
Fetch flow names and project name

### DISCUSSION THREAD

**Discussion by najoshi7** [PrefectHQ/prefect#3579] [Q&A] [answered]
Hello prefect community,

I have two questions:
1. How to find project-name of a flow? I tried following query in my Python program:
    ```
query = prefect.utilities.graphql.parse_graphql({
    'query': {
        'flow': {
            'id',
            'name',
            'project_name'   <----------- what is the field name for fetching project name for flow?
        }
    }
})
```
But the above query doesn't work. What should be the field name for fetching the name of project? 'project_name' or something else? I tried with 'project name' and 'project' also. But it doesn't work saying following error:
"requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://localhost:4200/graphql"

2. How to list flows only for a particular given project e.g. "project123"?

Thanks.

**Comment by joshmeek** [comment]
Try using the Prefect `Client` to grab information from the GraphQL API.

1: Something like this to get a project name:

```python
from prefect import Client

client = Client()

query = {
    "query": {
        "flow": {
            "id": True,
            "name: True,
            "project": {
                "name": True
            }
        }
    }
}

client.graphql(query)
```

2. For performing queries like this the library has some utility functions for composing graphql queries in python:

```python
from prefect.client import Client
from prefect.utilities.graphql import with_args

client = Client()

query = {
    "query": {
        with_args(
            "flow",
            {
                "where": {"project": {"name": {"_eq": "Demo"}}}
            }
        ): {
            "name": True,
            "id": True
        }
    }
}

client.graphql(query)
```

**Comment by najoshi7** [comment, marked as answer]
Thank you very much.
It helped me.
